[GS-89]Automate authentication for GS-to-Sheet script user

naveen-nathan opened the pull request:
### Jira Ticket

<!-- Replace the field marked <ticket-id> with your ticket id -->
[Jira Ticket](https://dashbd.atlassian.net/browse/GS-89)

### Description

<!-- Briefly describe the feature being introduced. -->

Automate authentication for GS-to-Sheet script user

### Type of Change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [X] Breaking change (fix or feature that would change existing functionality)

### Changes

<!-- List the major changes made in this pull request. -->

### Testing

<!-- Describe how the feature has been tested, including both automated and manual testing strategies. -->

### Checklist

- [X] My branch name matches the format: `<ticket-id>/<brief-description-of-change>`
- [X] My PR name matches the format: `[<ticket-id>] <brief-description-of-change>`
- [ ] I have added doc-comments to all new functions ([JSDoc](https://jsdoc.app/) for JS and [Docstrings](https://peps.python.org/pep-0257/) for Python)
- [X] I have reviewed all of my code
- [X] My code only contains major changes related to my ticket

### Screenshots/Video

<!-- Include screenshots or video demonstrating the new feature, if applicable. -->

### Additional Notes

<!-- Any additional information or context relevant to this PR. -->

naveen-nathan committed:
Automate authentication for GS-to-Sheet script user

Mehul-Gandhi commented on oldScripts/gradescope_to_spreadsheet.py:
```diff
@@ -155,8 +155,10 @@ def retrieve_grades_from_gradescope(gradescope_client, assignment_id = ASSIGNMEN
 
 @deprecated
 def initialize_gs_client():
+    with open('gs_credentials.json', 'r') as credentials_file:
```
Please load credentials from a .env file instead because this is sensitive data. Add a .gitignore with .env added to the .gitignore. Everything else is good. For an example, look at the api folder in this repo

naveen-nathan replied:
Thanks for pointing me to the example in the api folder, Mehul! This is the first time I am seeing a python script access environment variables. Glad to learn!

naveen-nathan committed:
Retrieve Gradescope credentials from .env file

Mehul-Gandhi commented on oldScripts/gradescope_to_spreadsheet.py:
```diff
@@ -331,4 +328,12 @@ def populate_instructor_dashboard():
+    else:
+        populate_instructor_dashboard()
+
 main()
```
```suggestion
if __name__ == "__main__":
    main()
```
https://realpython.com/if-name-main-python/
See GradeView `dbcron\update_bins.py` for another example

Mehul-Gandhi replied:
This is just a suggestion. After you review this, I can approve

naveen-nathan replied:
Thanks for introducing me to if __name__ == "__main__", Mehul! Committed your suggestion.

naveen-nathan committed:
Add .env file to .gitignore

Mehul-Gandhi commented on .gitignore:
```diff
@@ -0,0 +1 @@
+.env
```
Where do you have your .env file stored in? It might make more sense to have the `.env` file within the `oldScripts` folder because we might have multiple `.env` files if we add more folders later in development, similar to GradeView. The gitignore could go here, or moved to `oldScripts`, but I suggest the `.env` should go in the `oldScripts` folder

naveen-nathan replied:
Absolutely right! Will change this to oldScripts/.env

Mehul-Gandhi replied:
Ping me on slack once you've resolved the comments

naveen-nathan replied:
Thanks for your review, Mehul! All comments resolved

naveen-nathan replied:
Just realized I didn't press the resolved buttons; will do that now.

Mehul-Gandhi commented on oldScripts/gradescope_to_spreadsheet.py:
```diff
@@ -41,11 +43,14 @@
 # The ASSIGNMENT_ID constant is for users who wish to generate a sub-sheet (not update the dashboard) for one assignment, passing it as a parameter.
 ASSIGNMENT_ID = (len(sys.argv) > 1) and sys.argv[1]
 
+load_dotenv()
```
We will need to add instructions in a README on what configurations should go in the `.env` file if users want to run this script. This can be a separate ticket.

naveen-nathan replied:
Noted!

naveen-nathan committed:
Adopt  if __name__ == __main__ idiom

At Mehul's request; he shared this link for context: https://realpython.com/if-name-main-python/

Co-authored-by: Mehul Gandhi <48500458+Mehul-Gandhi@users.noreply.github.com>

naveen-nathan committed:
Update .gitignore

Mehul-Gandhi reviewed (approved):
LGTM
